### PR TITLE
XW-1480 | Add new header to mobile article preview

### PR DESCRIPTION
**Pull request by Warkot**
## Links

* http://wikia-inc.atlassian.net/browse/XW-1480

## Description

Redesign of the top of the article implementation, header to be more precise, changes how the header + infobox is handled. This fixes the issue when no hero image and title was shown.

## Reviewers

@Wikia/x-wing @Wikia/west-wing

### CODE REVIEW

File: front/main/app/controllers/article-preview.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Ember from 'ember';
+export default Ember.Controller.extend({
+	application: Ember.inject.controller(),
+
+	heroImage: Ember.computed('model.media', function () {
```
Comment (Warkot) [inline front/main/app/controllers/article-preview.js]: What do you think is the best way to DRY this code? Because this is copied'n'pasted from here https://github.com/Wikia/mercury/blob/ea9276d9e55e755d0d8b102ed87a407e67ec539a/front/main/app/components/article-wrapper.js#L106.

File: front/main/app/controllers/article-preview.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+		let heroImage;
+
+		this.get('model.media.media').forEach((current) => {
+			if (current.hasOwnProperty('context') && current.context === 'infobox-hero-image') {
+				heroImage = current;
```
Comment (dianafa) [inline front/main/app/controllers/article-preview.js]: Maybe it's a micro-optimisation but we actually don't have to iterate over media array from the moment we found first hero image element.

Comment (Warkot) [inline front/main/app/controllers/article-preview.js]: I was thinking about it but 
> There's no built-in ability to `break` in `forEach`. To interrupt execution you would have to throw an exception of some sort. eg.

http://stackoverflow.com/questions/2641347/how-to-short-circuit-array-foreach-like-calling-break

😿

Comment (Warkot) [inline front/main/app/controllers/article-preview.js]: Something like this would work:
```javascript
heroImage: Ember.computed('model.media', function () {
	const media = this.get('model.media.media');

	if (Ember.isArray(media)) {
		for (let i = 0; i < media.length; i++) {
			if (media[i].hasOwnProperty('context') && media[i].context === 'infobox-hero-image') {
				return media[i];
			}
		}
	}

	return null;
})
```
What do you think?

Comment (Warkot) [inline front/main/app/controllers/article-preview.js]: Okay, I did it here https://github.com/Wikia/mercury/pull/2434/commits/8136683e3ef19e1a61e30bfd9e4b3ab1f4f8dca4. 😼

File: front/main/app/mixins/hero-image.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create({
+	heroImage: Ember.computed('model.media', function () {
+		const media = this.get('model.media.media');
+
+		if (Ember.isArray(media)) {
+			for (let i = 0; i < media.length; i++) {
+				if (media[i].hasOwnProperty('context') && media[i].context === 'infobox-hero-image') {
```
Comment (dianafa) [inline front/main/app/mixins/hero-image.js]: hmm if we iterate over media array, `media[i]` will always exist, thus we don't need to check here if hasOwnProp - in the worst case `media[i].context` just will be undefined, what do you think?

File: front/main/app/mixins/hero-image.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+					return media[i];
+				}
+			}
+		}
+
+		return null;
```
Comment (dianafa) [inline front/main/app/mixins/hero-image.js]: do we have to explicitly return null here?

File: front/main/app/controllers/article-preview.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import Ember from 'ember';
+import HeroImageMixin from '../mixins/hero-image';
+
+export default Ember.Controller.extend(HeroImageMixin, {
+	application: Ember.inject.controller(),
```
Comment (rogatty) [inline front/main/app/controllers/article-preview.js]: What's that for?

File: front/main/app/mixins/hero-image.js (hunk 3 above)
Comment (Warkot) [inline front/main/app/mixins/hero-image.js]: Agreed. I copy-pasted to much. :+1:

File: front/main/app/styles/component/wikia-ui-components/_wiki-page-header.scss [hunk 6]
```diff
@@ -75,7 +75,7 @@
 }
 
 // component modifier-related changes to elements
-.wiki-page-header.has-action .wiki-page-header__wrapper {
+.wiki-page-header .wiki-page-header__wrapper {
```
Comment (rogatty) [inline front/main/app/styles/component/wikia-ui-components/_wiki-page-header.scss]: Why was this removed? We already have rules for `.wiki-page-header__wrapper` above.

File: front/main/app/mixins/hero-image.js (hunk 4 above)
Comment (Warkot) [inline front/main/app/mixins/hero-image.js]: No. It just felt right to me to always return something if I already return other values. I'll drop it.

Comment (rogatty) [inline front/main/app/mixins/hero-image.js]: I like `null` better than `undefined` :)

**Comment by rogatty** [comment]
JS LGTM but I'd prefer to let @vforge speak about SCSS.

File: front/main/app/styles/component/wikia-ui-components/_wiki-page-header.scss [hunk 7]
```diff
@@ -9,6 +9,50 @@
...
 	background: $color-blue-light;
 	color: #fff;
+
+	.wiki-page-header__wrapper {
+		display: flex;
+	}
+
+	// component modifier-related changes to elements
+	&.has-hero-image {
+		background-color: #fff;
```
Comment (hakubo) [inline front/main/app/styles/component/wikia-ui-components/_wiki-page-header.scss]: background: {
color:
position:
...
}

**Comment by vforge** [comment]
@rogatty Talked offline, LQGTM
